### PR TITLE
feat: adapt `consistency_checker` tests for running both in Validium mode and Rollup mode

### PR DESCRIPTION
**Pull request by jordibonet-lambdaclass**
# Failing tests
## `build_commit_tx_input_data_is_correct` 
```
assertion `left == right` failed
  left: Tuple([Uint(1), Uint(1), Uint(21), FixedBytes([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1]), Uint(0), FixedBytes([197, 210, 70, 1, 134, 247, 35, 60, 146, 126, 125, 178, 220, 199, 3, 192, 229, 0, 182, 83, 202, 130, 39, 59, 123, 250, 216, 4, 93, 133, 164, 112]), FixedBytes([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]), FixedBytes([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]), Bytes([]), Bytes([0])])
 right: Tuple([Uint(1), Uint(1), Uint(21), FixedBytes([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1]), Uint(0), FixedBytes([197, 210, 70, 1, 134, 247, 35, 60, 146, 126, 125, 178, 220, 199, 3, 192, 229, 0, 182, 83, 202, 130, 39, 59, 123, 250, 216, 4, 93, 133, 164, 112]), FixedBytes([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]), FixedBytes([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]), Bytes([])])
```
## `normal_checker_function` `checker_processes_pre_boojum_batches` `checker_functions_after_snapshot_recovery`
```
thread 'consistency_checker::tests::normal_checker_function::case_00' panicked at core/lib/zksync_core/src/consistency_checker/tests/mod.rs:369:68:
called `Option::unwrap()` on a `None` value
```
It fails in this line: 
```rust
let checked_batch = l1_batch_updates_receiver.recv().await.unwrap();
```
# Run the tests 

To run the tests, you can execute the following command:
```sh
zk test rust --package zksync_core --lib -j 1 -- consistency_checker::tests
```

# Fix tests
Failed test errors occur when comparing the result of the contract deployed using the "commitBaches" method with the `ValidiumModeL1BatchCommitDataGenerator`. 

This is solved by adding the following change:

https://github.com/lambdaclass/zksync-era/blob/159225b355caf0622d48a83bdb41e73856753387/core/lib/types/src/l1_batch_commit_data_generator.rs#L25-L29

I believe this change has been commented on at the last validium meeting by @ilitteri .

cc @mationorato @ilitteri

### CODE REVIEW

**Comment by jordibonet-lambdaclass** [comment]
build_commit_tx_input_data_is_correct fails
```
assertion `left == right` failed
  left: Tuple([Uint(1), Uint(1), Uint(21), FixedBytes([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1]), Uint(0), FixedBytes([197, 210, 70, 1, 134, 247, 35, 60, 146, 126, 125, 178, 220, 199, 3, 192, 229, 0, 182, 83, 202, 130, 39, 59, 123, 250, 216, 4, 93, 133, 164, 112]), FixedBytes([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]), FixedBytes([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]), Bytes([]), Bytes([0])])
 right: Tuple([Uint(1), Uint(1), Uint(21), FixedBytes([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1]), Uint(0), FixedBytes([197, 210, 70, 1, 134, 247, 35, 60, 146, 126, 125, 178, 220, 199, 3, 192, 229, 0, 182, 83, 202, 130, 39, 59, 123, 250, 216, 4, 93, 133, 164, 112]), FixedBytes([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]), FixedBytes([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]), Bytes([])])
```

**Review by ilitteri — CHANGES_REQUESTED** [review]
Great work! Some comments:

- Let's move the auxiliary functions (the abstractions) to a `test_helpers` module. If I'm not wrong, by doing so we an get rid of the prefix `_` and call the methods like `test_helpers::xxx` after importing the module `test_helpers`. 
- Let's add how to run the tests in the PR's description and title (the actual title tells more about *how* it was solved rather than *what* was solved).

EDIT: Regarding your suggested solution, it is exactly what we need and it is probably the solution we're looking for. I'll file an issue for solving that.

File: core/lib/zksync_core/src/consistency_checker/tests/mod.rs [hunk 1]
```diff
@@ -9,19 +10,20 @@ use zksync_dal::StorageProcessor;
...
 use super::*;
-use crate::{
-    genesis::{ensure_genesis_state, GenesisParams},
-    utils::testonly::{create_l1_batch, create_l1_batch_metadata},
-};
+use crate::utils::testonly::{create_l1_batch, create_l1_batch_metadata};
 
 /// **NB.** For tests to run correctly, the returned value must be deterministic (i.e., depend only on `number`).
-fn create_l1_batch_with_metadata(number: u32) -> L1BatchWithMetadata {
+pub(crate) fn create_l1_batch_with_metadata(number: u32) -> L1BatchWithMetadata {
```
Comment (ilitteri) [inline core/lib/zksync_core/src/consistency_checker/tests/mod.rs:26]: Is it necessary for this function and the others to be public for the crate?

Comment (ilitteri) [reply core/lib/zksync_core/src/consistency_checker/tests/mod.rs:26]: Nevermind, I just saw why

File: core/lib/zksync_core/src/consistency_checker/tests/mod.rs [hunk 2]
```diff
@@ -298,222 +285,54 @@ async fn normal_checker_function(
...
+    test_helpers::checker_functions_after_snapshot_recovery(
+        delay_batch_insertion,
+        Arc::new(ValidiumModeL1BatchCommitDataGenerator {}),
+    )
+    .await;
 }
 
 #[derive(Debug, Clone, Copy)]
-enum IncorrectDataKind {
+pub enum IncorrectDataKind {
```
Comment (ilitteri) [inline core/lib/zksync_core/src/consistency_checker/tests/mod.rs:335]: Should this be public for the crate also?

File: core/lib/zksync_core/src/consistency_checker/tests/mod.rs [hunk 3]
```diff
@@ -191,7 +178,7 @@ fn extracting_commit_data_for_pre_boojum_batch() {
 }
 
 #[derive(Debug, Clone, Copy)]
-enum SaveAction<'a> {
+pub enum SaveAction<'a> {
```
Comment (ilitteri) [inline core/lib/zksync_core/src/consistency_checker/tests/mod.rs:181]: Ditto

File: core/lib/zksync_core/src/consistency_checker/tests/mod.rs (hunk 2 above)
Comment (jordibonet-lambdaclass) [reply core/lib/zksync_core/src/consistency_checker/tests/mod.rs:335]: I tried, but the test_helpers module didn't recognize the function, so I put `pub` without `(crate)` in this case

**Review by ilitteri — APPROVED** [review]
Nice work refactoring the helper functions. We could do the same for `create_l1_batch_with_metadata`, `create_pre_boojum_l1_batch_with_metadata`, `build_commit_tx_input_data`, `create_mock_checker` and for every definition that is not a test, but let's move on with this and wait for their review.

File: core/lib/zksync_core/src/consistency_checker/tests/mod.rs (hunk 2 above)
Comment (jordibonet-lambdaclass) [reply core/lib/zksync_core/src/consistency_checker/tests/mod.rs:335]: I have seen the problem, I'm correcting it. Thank you!

**Pull request merged**